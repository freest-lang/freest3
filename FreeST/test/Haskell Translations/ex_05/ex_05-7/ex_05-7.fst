-- V exercise 7

data IntList = End | List Int IntList

dropUntil' : (Int -> Bool) -> IntList -> IntList
dropUntil' f list =
    case list of {
        End -> End,
        List x rest -> 
            if f x 
                then List x rest
                else dropUntil' f rest
    }

isPositive : Int -> Bool
isPositive x = x > 0

list : IntList
list = List (-4) (List 0 (List (-8) (List 3 (List (-2) (List (-5) (List 3 End))))))

main : IntList 
main = dropUntil' isPositive list
-- result = List 3 (List (-2) (List (-5) (List 3 End)))
