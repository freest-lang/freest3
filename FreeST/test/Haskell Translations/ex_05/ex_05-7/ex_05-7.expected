List 3 (List (-2) (List (-5) (List 3 End)))